useragent/runtime.go

```diff
@@ -0,0 +1,29 @@
+package useragent
+
+import (
+	"os"
+	"sync"
+)
+
+var runtimeOnce sync.Once
+var runtimeVersion string
+
+// DisableRuntimeCaching is a flag to disable caching of runtime version.
+// This is useful for testing.
+var DisableRuntimeCaching bool
+
+func getRuntimeVersion() string {
+	v := os.Getenv("DATABRICKS_RUNTIME_VERSION")
+	return Sanitize(v)
+}
+
+func Runtime() string {
+	if DisableRuntimeCaching {
+		return getRuntimeVersion()
+	}
+
+	runtimeOnce.Do(func() {
+		runtimeVersion = getRuntimeVersion()
+	})
+	return runtimeVersion
+}
```

Comment:
Like the CI/CD test, this one can have a companion test where you verify that a few select runtime version strings are correctly returned/sanitized and are valid. Then the e2e test can just verify that it is set and contains something and not iterate over all options.

Also removes the need for exported symbol for `DisableRuntimeCaching`.

Reply:
done

Reply:
The end to end test is failing because of the caching. Does it make sense to keep the exported flag?

Reply:
What do we do for the CI/CD one?

Reply:
We originally did had the e2e test for the CI/CD one. Also the tests are run from GitHub actions so another one of our tests as a side effect asserted on the CICD useragent. (`TestNonJSONResponseIncludedInError`)

Reply:
Done. Added a `ClearCache` method and also added an e2e test for the cicd user agent.